src/Adapter/Product/ProductDataProvider.php

```diff
@@ -37,10 +37,15 @@ class ProductDataProvider
     /**
      * Get a new ProductCore instance
      *
-     * @return object product
+     * @param null $idProduct
+     * @return Product
      */
-    public function getProductInstance()
+    public function getProductInstance($idProduct = null)
     {
+        if ($idProduct) {
+            return new Product($idProduct);
+        }
```

Comment:
Too bad there is currently no Entity for the Product class, this would be really better. It seems we just reproduce the legacy errors here :upside_down_face: